### PR TITLE
mapmathing: - added to services docs; - fix api-ref docs

### PR DESCRIPTION
**Pull request by RingerJK**
Closes https://github.com/mapbox/mapbox-java/issues/1398

### CODE REVIEW

File: services/build.gradle [hunk 1]
```diff
@@ -6,14 +6,15 @@ sourceSets {
                          '../services-optimization/src/main/java', '../services-staticmap/src/main/java',
                          '../services-speech/src/main/java', '../services-tilequery/src/main/java',
                          '../services-route-tiles/src/main/java', '../services-directions-refresh/src/main/java',
-                         '../services-isochrone/src/main/java']
+                         '../services-isochrone/src/main/java', '../services-matching/src/main/java']
```
Comment (RingerJK) [inline services/build.gradle:9]: include `mapmatching` to `services` api-ref docs  

cc @Guardiola31337

File: services/build.gradle [hunk 2]
```diff
@@ -6,14 +6,15 @@ sourceSets {
...
 }
 
 dependencies {
     api project(":services-core")
     api project(":services-geojson")
     api project(":services-directions-models")
     api project(":services-directions-refresh-models")
+    api project(":services-matching")
```
Comment (Guardiola31337) [inline services/build.gradle:17]: Why are we adding this but `services-isochrone` or `services-speech` (to name a couple) are not in here? 🤔

Comment (RingerJK) [reply services/build.gradle]: removed, it's not affecting docs

Comment (Guardiola31337) [reply services/build.gradle]: At some point we should revisit this (what we're including as part of `services`). Could you cut a ticket when you have a chance so we don't forget? 🙏

**Review by Guardiola31337 — APPROVED** [review]

File: services/build.gradle (hunk 2 above)
Comment (RingerJK) [reply services/build.gradle]: done https://github.com/mapbox/mapbox-java/issues/1401

Comment (Guardiola31337) [reply services/build.gradle]: Thanks 🙇‍♂️

File: services/build.gradle (hunk 1 above)
Comment (zugaldia) [reply services/build.gradle:9]: @RingerJK @Guardiola31337  When will we see the new map matching docs on https://docs.mapbox.com/android/java/api-reference/?

Comment (Guardiola31337) [reply services/build.gradle:9]: They're already available under [mapbox-java-services](https://docs.mapbox.com/android/java/api/libjava-services/6.4.0/) e.g. [`MapMatchingService`](https://docs.mapbox.com/android/java/api/libjava-services/6.4.0/com/mapbox/api/matching/v5/MapMatchingService.html) 🚀

**Pull request merged**